server/src/main/java/io/deephaven/server/object/ObjectServiceGrpcImpl.java

```diff
@@ -89,112 +205,71 @@ private FetchObjectResponse serialize(String expectedType, SessionState state, O
+                export.release();
             } catch (Throwable inner) {
                 t.addSuppressed(inner);
             }
         }
     }
 
-    private static boolean referenceEquality(Object t, Object u) {
-        return t == u;
-    }
-
-    final class ExportCollector implements Exporter {
-
+    private final class PluginMessageSender implements ObjectType.MessageStream {
+        private final StreamObserver<StreamResponse> responseObserver;
         private final SessionState sessionState;
-        private final Thread thread;
-        private final List<ReferenceImpl> references;
 
-        public ExportCollector(SessionState sessionState) {
-            this.sessionState = Objects.requireNonNull(sessionState);
-            this.thread = Thread.currentThread();
-            this.references = new ArrayList<>();
-        }
+        private boolean closed;
 
-        public List<ReferenceImpl> refs() {
-            return references;
+        public PluginMessageSender(StreamObserver<StreamResponse> responseObserver, SessionState sessionState) {
+            this.responseObserver = responseObserver;
+            this.sessionState = sessionState;
         }
 
         @Override
-        public Optional<Reference> reference(Object object, boolean allowUnknownType, boolean forceNew) {
-            return reference(object, allowUnknownType, forceNew, ObjectServiceGrpcImpl::referenceEquality);
-        }
+        public void onData(ByteBuffer message, Object[] references) {
+            Data.Builder payload = Data.newBuilder().setPayload(ByteString.copyFrom(message));
 
-        @Override
-        public Optional<Reference> reference(Object object, boolean allowUnknownType, boolean forceNew,
-                BiPredicate<Object, Object> equals) {
-            if (thread != Thread.currentThread()) {
-                throw new IllegalStateException("Should only create references on the calling thread");
-            }
-            if (!forceNew) {
-                for (ReferenceImpl reference : references) {
-                    if (equals.test(object, reference.export.get())) {
-                        return Optional.of(reference);
-                    }
+            List<ExportObject<?>> exports = new ArrayList<>();
+            try {
+                for (Object reference : references) {
+                    final String type = typeLookup.type(reference).orElse(null);
+                    final ExportObject<?> exportObject = sessionState.newServerSideExport(reference);
+                    exports.add(exportObject);
+                    TypedTicket typedTicket = ticketForExport(exportObject, type);
+                    payload.addTypedExportIds(typedTicket);
                 }
+            } catch (RuntimeException | Error t) {
+                cleanup(exports, t);
+                throw t;
+            } catch (Throwable t) {
+                cleanup(exports, t);
+                throw new RuntimeException(t);
```

Comment:
I don't think this catch is serving any purpose.

Reply:
As above, it guards against sneaky-throws, which (since we don't allow that pattern) would by definition be external java plugins. The original code that this is based on supported Throwable and wrapped in this way, I assumed for the same reason? Or just as a quick and dirty way to avoid the `RuntimeException | Error`?